File: openmole-ui-ide/openmole-ide-ant/org.openmole.ui.ide/src/org/openmole/ui/ide/workflow/model/IObjectModelUI.java
/*
 *  Copyright (C) 2010 Mathieu Leclaire <mathieu.leclaire@openmole.fr>
 *
 *  This program is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU Affero General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package org.openmole.ui.ide.workflow.model;

import org.openmole.commons.aspect.eventdispatcher.IObjectChangedAsynchronousListener;

/**
 *
 * @author Mathieu Leclaire <mathieu.leclaire@openmole.fr>
 */
public interface IObjectModelUI <T> extends IObjectChangedAsynchronousListener<T>{
    @Override
    void objectChanged(T obj);
    void updateData();
    void setName(String name);
    String getName();
    String getModelGroup();
    String getCategory();
}

File: platform/gui/src/main/scala/org/openmole/ui/ide/workflow/model/IObjectModelUI.scala
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package org.openmole.ui.ide.workflow.model

import org.openmole.misc.eventdispatcher.IObjectListener

trait IObjectModelUI[T] extends IObjectListener[T] {
  def objectChanged(obj: T)
  
  def updateData()
  
 // def getCategory(): String
}
//
//<T> extends IObjectListener<T>{
////public interface IObjectModelUI <T>{
//    void objectChanged(T obj);
//    void updateData();
//    String getModelGroup();
//    String getCategory();